zerver/actions/realm_settings.py

```diff
@@ -104,6 +105,43 @@ def do_set_realm_property(
         update_users_in_full_members_system_group(realm, acting_user=acting_user)
 
 
+@transaction.atomic(durable=True)
+def do_set_realm_permission_group_settings(
+    realm: Realm, name: str, user_group: UserGroup, *, acting_user: Optional[UserProfile]
+) -> None:
+    """Takes in a realm object, the name of an attribute to update, the
+    user_group to update and and the user who initiated the update.
+    """
+
+    old_user_group_id = getattr(realm, name)
```

Comment:
Let's change `name` to `setting_name`, and have an `assert setting_name in REALM_PERMISSION_GROUP_SETTING_ID_TO_GROUP_SETTING` line.